### PR TITLE
Services and stores

### PR DESCRIPTION
**Pull request by pavkahanov**

### CODE REVIEW

File: src/services/ContractService/ContractService.js [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+
+  /**
+   * calling contract method
+   * @param {string} method method, which will be called
+   * @param {string} from address of caller
+   */
+  async callMethod(method, from, params) {
+    const data = await this.contract.methods[method](params).call({ from });
+    return data;
+  }
```
Comment (masteralx) [inline src/services/ContractService/ContractService.js]: params is not defined in params list

File: src/services/ContractService/ContractService.js [hunk 2]
```diff
@@ -1,3 +1,29 @@
-export default class ContractService {
+/**
+ * Class for forming transactions
+ */
+class ContractService {
```
Comment (masteralx) [inline src/services/ContractService/ContractService.js:4]: What contract methods will you need?

File: src/services/WalletService/WalletService.js [hunk 3]
```diff
@@ -1,3 +1,49 @@
...
+   * @param {string} encryptedWallet
+   */
+  writeWalletToFile(encryptedWallet) {
+
+  }
+
+  /**
+   * Creates new wallet
+   * @param {string} password - combination of symbols which will be allow decode wallet
+   * @returns {array} Encrypted wallet and seed
```
Comment (masteralx) [inline src/services/WalletService/WalletService.js]: array of what?

File: src/services/WalletService/WalletService.js [hunk 4]
```diff
@@ -1,3 +1,49 @@
-export default class WalletService {
+/** Service for working with wallets */
+class WalletService {
+  /**
+   * Decrypts wallet
+   * @param {string} url path to wallet
+   * @param {string} password password for decrypting
+   */
```
Comment (masteralx) [inline src/services/WalletService/WalletService.js:8]: returns?

File: src/services/WalletService/WalletService.js [hunk 5]
```diff
@@ -1,3 +1,49 @@
...
+  readWalletFromFile(url, password) {
 
+  }
+
+  /**
+   * Write encrypted wallet to file
+   * @param {string} encryptedWallet
+   */
```
Comment (masteralx) [inline src/services/WalletService/WalletService.js:17]: returns?

File: src/services/WalletService/WalletService.js [hunk 6]
```diff
@@ -1,3 +1,49 @@
...
+   * Checking wallet for creating txDaat
+   * @param {string} wallet encryptedWallet
+   * @param {string} password password
+   * @return {bool} is password correct
+   */
+  checkPassword(wallet, password) {
+
+  }
 }
+const walletService = new WalletService();
```
Comment (masteralx) [inline src/services/WalletService/WalletService.js]: singleton?

File: src/services/Web3Service/Web3Service.js [hunk 7]
```diff
@@ -1,3 +1,77 @@
...
+    this.web3.setProvider(provider);
+  }
+
+  /**
+   * sets wallet to web3 by private key
+   * @param {string} privateKey private key
+   */
+  setWallet(privateKey) {
+    this.web3.eth.accounts.wallet.add(privateKey);
+  }
```
Comment (masteralx) [inline src/services/Web3Service/Web3Service.js]: For what?

File: src/services/Web3Service/Web3Service.js [hunk 8]
```diff
@@ -1,3 +1,77 @@
...
+
+  /**
+   * Sending transaction to contract
+   * @param {string} txData Raw transaction (without 0x)
+   * @param {string} from User, who send transaction
+   */
```
Comment (masteralx) [inline src/services/Web3Service/Web3Service.js:47]: returns?

File: src/services/Web3Service/Web3Service.js [hunk 9]
```diff
@@ -1,3 +1,77 @@
...
+  sendSignedTransaction(txData) {
+    this.address2nonce[from] += 1;
```
Comment (masteralx) [inline src/services/Web3Service/Web3Service.js:49]: When it decrements?

File: src/stores/AppStore/AppStore.js [hunk 10]
```diff
@@ -1,3 +1,47 @@
...
+import ProjectStore from '../ProjectStore';
+
 export default class AppStore {
+  @observable walletList = [];
+
+  @observable projectList = [];
+
+  @observable userStore = new UserStore()
+
+  @observable projectStore;
```
Comment (masteralx) [inline src/stores/AppStore/AppStore.js]: Why is it here?

File: src/stores/ProjectStore/ProjectStore.js [hunk 11]
```diff
@@ -1,3 +1,76 @@
...
 
+/**
+ * Class implements whole project
+ */
+class ProjectStore {
+  @observable projectAddress = ''
+
+  @observable prepared = 0;
+
+  @observable votingStats = {
```
Comment (masteralx) [inline src/stores/ProjectStore/ProjectStore.js]: observable?

Comment (masteralx) [inline src/stores/ProjectStore/ProjectStore.js]: stats? states?

File: src/stores/QuestionStore/QuestionStore.js [hunk 12]
```diff
@@ -1,3 +1,67 @@
-export default class QuestionStore {
+import { observable, action, computed } from 'mobx';
+import Question from './entities/Question';
 
+/**
+ * Contains methods for working
+ */
+class QuestionStore {
+  @observable _questions;
+
```
Comment (masteralx) [inline src/stores/QuestionStore/QuestionStore.js:9]: extra line

File: src/stores/QuestionStore/entities/Question.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+   * @return {object} data about question
+   */
+  selectQuestionForVoting() {
+    const {
+      id, groupId, caption, text, params,
+    } = this;
+    return {
+      id, groupId, caption, text, params,
+    };
+  }
```
Comment (masteralx) [inline src/stores/QuestionStore/entities/Question.js]: Should it be here?

File: src/stores/UserStore/UserStore.js [hunk 14]
```diff
@@ -1,3 +1,57 @@
...
+  @action setEncryptedWallet = (wallet) => {
+    this.encryptedWallet = wallet;
+  }
+
+  /**
+   * checking password by wallet service
+   * @param password password for decoding
+   * @return {bool} is password correct
+   */
+  @action checkPassword = (password) => {
```
Comment (masteralx) [inline src/stores/UserStore/UserStore.js]: Why is it arrow function?

File: src/stores/UserStore/UserStore.js [hunk 15]
```diff
@@ -1,3 +1,57 @@
...
+  }
+
+  /**
+   * Signing transactions with private key
+   * @function
+   * @param {string} data rawTx
+   * @param {string} password password which was used to encode Keystore V3
+   * @return Signed TX data
+   */
+  @action singTransaction = (data, password) => {
```
Comment (masteralx) [inline src/stores/UserStore/UserStore.js]: Why is it arrow function?

File: src/stores/UserStore/UserStore.js [hunk 16]
```diff
@@ -1,3 +1,57 @@
...
+
+  }
+
+  /**
+   * Sending transaction from user
+   * @function
+   * @param {string} txData Raw transaction
+   */
+  @action sendTransaction = (txData) => {
```
Comment (masteralx) [inline src/stores/UserStore/UserStore.js]: Why is it arrow function?

**Review by masteralx — CHANGES_REQUESTED** [review]

File: src/stores/QuestionStore/entities/Question.js (hunk 13 above)
Comment (pavkahanov) [reply src/stores/QuestionStore/entities/Question.js]: Removed, because we have getQuestionById

File: src/stores/AppStore/AppStore.js (hunk 10 above)
Comment (pavkahanov) [reply src/stores/AppStore/AppStore.js]: Moved to Rootstore

**Pull request merged**